Initial tidy up

hannahhaken opened the pull request:
Deleted core
Status enum moved to separate class
Remove [Required] syntax
`DueDate` to `DueAt`
Added a blazor project

I haven't renamed 'Domain' because when I did, it didn't look like a simple refactor rename....unless I'm wrong.

hannahhaken committed:
Add initial Blazor project structure with required assets

hannahhaken committed:
Remove Core project and restructure Domain.

The Core project, including its configuration and code, was removed entirely.

Domain, where new classes (User, TodoStatus) were added, and TodoItem was reorganized.

Updated the solution structure to include a new Blazor project.

garethbrownsb commented on Domain/User/User.cs:
```diff
@@ -0,0 +1,20 @@
+
+    public required string Email { get; set; } = string.Empty;
+
+    public required string Password { get; set; } = string.Empty;
```
Initializing these to string.Empty is not necessary. We'll discuss.

garethbrownsb commented on Domain/Todo/TodoItem.cs:
```diff
@@ -1,33 +1,20 @@
     public string? Description { get; set; }
 
-    public TodoStatus Status { get; set; }
+    public TodoStatus Status { get; set; } = TodoStatus.Due;
```
This doesn't need to be initialized

hannahhaken committed:
Removed initialising of certain User and TodoItem properties

hannahhaken committed:
Adds constructor-based initialisation for TodoItem and user

hannahhaken committed:
Defines domain behaviour for User: methods for updating fields and tracking times of changes

hannahhaken committed:
Defines domain behaviour for TodoItem: methods for updating fields and tracking times of changes

hannahhaken committed:
Add database initialization with SQLite

Introduce `DatabaseInitializer` to set up SQLite tables for `Users` and `TodoItems`.

Update `Program.cs` to initialize the database on app start. Adjust host address in `TodoApi.http`, and remove unused `RootNamespace` configuration from `Api.csproj`.

hannahhaken committed:
Refactor and rename solution for consistent naming.

Renamed project, namespaces, files, and directories to align with the "TodoApp" naming convention for improved consistency and clarity.

 Refactored `TodoItem` class structure and moved domain logic to `TodoApp.Core`.

 Added unit and integration test projects.

hannahhaken committed:
Removes AI generated documentation from TodoItem class.

hannahhaken committed:
Add database migrations and refactor initialization logic

Introduced DbUp for managing database migrations and removed inline SQL table creation.

 Updated appsettings and added required dependencies to support configuration and migration handling.

hannahhaken committed:
Add TodoController with CRUD operations and DB connection

Introduced a `TodoController` to provide CRUD functionality for managing todo items using Dapper and SQLite. Registered `IDbConnection` as a scoped service for database interactions. Updated project configuration to reference core project.

hannahhaken committed:
Add TodoApp.Playground project with string extensions and tests

Introduce a new TodoApp.Playground project for experimenting with features. Add string extension methods for appending and tracking parameter manipulation, along with corresponding unit tests to validate functionality. Update the solution file to include the new project.
